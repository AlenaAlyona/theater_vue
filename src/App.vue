<template>
  <div class="container">
    <h3 class="title">Theater Layout</h3>
    <Layout />
  </div>
</template>

<script>
import Layout from "./components/Layout.vue";

export default {
  name: "App",
  components: {
    Layout,
  },
};
</script>

<style>
#app {
  font-family: Work Sans, Helvetica Neue, Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  color: #2c3e50;
  background-image: linear-gradient(
    -55deg,
    #6d3076,
    #ff5252 83%,
    #ff5852 88%,
    #ff6b52
  );
  display: flex;
  height: 100%;
  align-items: center;
  flex-direction: column;
  overflow: auto;
}

html,
body {
  height: 100%;
  margin: 0;
  padding: 0;
  text-size-adjust: 100%;
}

.title {
  font-size: 4rem;
  line-height: 1;
  font-weight: 600;
  text-transform: lowercase;
  margin: 0;
  color: #fff;
  margin-top: 8rem;
  margin-bottom: 4rem;
}
</style>
