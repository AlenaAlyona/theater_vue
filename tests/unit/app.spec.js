import App from "../../src/App.vue";
import { mount } from "@vue/test-utils";

describe("App.vue", () => {
  test("mount App.vue", () => {
    const LayoutStub = {
      props: ["layout", "groups"],
    };

    const wrapper = mount(App, {
      stubs: {
        Layout: LayoutStub,
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
