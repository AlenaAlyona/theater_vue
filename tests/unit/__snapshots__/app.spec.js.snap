// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue mount App.vue 1`] = `
<div>
  <h3 class="title">Theater Layout</h3>
  <div class="layoutContainer">
    <div>
      <div class="sectionName">
        <div class="sectionContainer">
          <div class="sectionTitle">main hall</div>
          <div class="sectionLayout">
            <table>
              <thead>
                <tr>
                  <th class="thRow">row</th>
                  <th colspan="5">seats</th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <th scope="row" class="sectionRow">1</th>
                  <div class="seatsRow">
                    <td style="background: rgb(225, 221, 224);" class="sectionSeat"></td>
                    <td style="background: rgb(225, 221, 224);" class="sectionSeat"></td>
                    <td style="background: rgb(31, 31, 31);" class="sectionSeat">gr1</td>
                    <td style="background: rgb(31, 31, 31);" class="sectionSeat">gr1</td>
                  </div>
                </tr>
                <tr>
                  <th scope="row" class="sectionRow">2</th>
                  <div class="seatsRow">
                    <td style="background: rgb(225, 221, 224);" class="sectionSeat"></td>
                    <td style="background: rgb(225, 221, 224);" class="sectionSeat"></td>
                    <td style="background: rgb(31, 31, 31);" class="sectionSeat">gr2</td>
                    <td style="background: rgb(31, 31, 31);" class="sectionSeat">gr2</td>
                    <td style="background: rgb(31, 31, 31);" class="sectionSeat">gr1</td>
                    <td style="background: rgb(31, 31, 31);" class="sectionSeat">gr1</td>
                  </div>
                </tr>
                <tr>
                  <th scope="row" class="sectionRow">3</th>
                  <div class="seatsRow">
                    <td style="background: rgb(156, 156, 156);" class="sectionSeat"></td>
                    <td style="background: rgb(156, 156, 156);" class="sectionSeat"></td>
                    <td style="background: rgb(156, 156, 156);" class="sectionSeat"></td>
                    <td style="background: rgb(156, 156, 156);" class="sectionSeat"></td>
                    <td style="background: rgb(156, 156, 156);" class="sectionSeat"></td>
                    <td style="background: rgb(156, 156, 156);" class="sectionSeat"></td>
                  </div>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
      <div class="sectionName">
        <div class="sectionContainer">
          <div class="sectionTitle">balcony</div>
          <div class="sectionLayout">
            <table>
              <thead>
                <tr>
                  <th class="thRow">row</th>
                  <th colspan="5">seats</th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <th scope="row" class="sectionRow">1</th>
                  <div class="seatsRow">
                    <td style="background: rgb(225, 221, 224);" class="sectionSeat"></td>
                    <td style="background: rgb(225, 221, 224);" class="sectionSeat"></td>
                    <td style="background: rgb(225, 221, 224);" class="sectionSeat"></td>
                    <td style="background: rgb(225, 221, 224);" class="sectionSeat"></td>
                  </div>
                </tr>
                <tr>
                  <th scope="row" class="sectionRow">2</th>
                  <div class="seatsRow">
                    <td style="background: rgb(225, 221, 224);" class="sectionSeat"></td>
                    <td style="background: rgb(225, 221, 224);" class="sectionSeat"></td>
                    <td style="background: rgb(225, 221, 224);" class="sectionSeat"></td>
                    <td style="background: rgb(225, 221, 224);" class="sectionSeat"></td>
                    <td style="background: rgb(225, 221, 224);" class="sectionSeat"></td>
                    <td style="background: rgb(225, 221, 224);" class="sectionSeat"></td>
                  </div>
                </tr>
                <tr>
                  <th scope="row" class="sectionRow">3</th>
                  <div class="seatsRow">
                    <td style="background: rgb(156, 156, 156);" class="sectionSeat"></td>
                    <td style="background: rgb(156, 156, 156);" class="sectionSeat"></td>
                    <td style="background: rgb(31, 31, 31);" class="sectionSeat">gr3</td>
                    <td style="background: rgb(31, 31, 31);" class="sectionSeat">gr3</td>
                    <td style="background: rgb(156, 156, 156);" class="sectionSeat"></td>
                    <td style="background: rgb(156, 156, 156);" class="sectionSeat"></td>
                  </div>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
